docs/index.asciidoc

```diff
@@ -23,24 +23,293 @@ include::{include_path}/plugin_header.asciidoc[]
+<4> Determines whether the event source is a queue or a topic, set to `true` for topic, `false` for queue.
+<5> Credentials to use when connecting to the JMS provider, if required.
+<6> Keystore and Truststore to use when connecting to the JMS provider, if required.
+<7> Parts of the JMS Message to include in the event - headers, properties and the message body can be included or
+    excluded from the event.
+<8> Determines whether the JMSTimestamp header is used to populate the `@timestamp` field.
+<9> Message selector: Use this to filter messages to be processed. The whole selector query should be double-quoted,
+    string property values should be single quoted, and numeric property vaues should not be quoted.
+    See JMS provider documentation for exact syntax.
+<10> Additional settings that are set directly on the ConnectionFactory object can be added here.
+<11> List of jars required by the JMS provider. Paths should be the fully qualified location of all jar files required
+     by the JMS provider. This list may also include dependent jars as well as specific jars from the JMS provider.
+
+===== Sample Configuration for JNDI Connection
+[source,ruby]
+--------------------------------------------------
+ input {
+    jms {
+        include_header => false <1>
+        include_properties => false <1>
+        include_body => true <1>
+        use_jms_timestamp => false <1>
+        interval => 10 <1>
+        destination => "myqueue" <1>
+        pub_sub => false <1>
+        jndi_name => /jms/cf/default <2>
+        jndi_context => {
+         'java.naming.factory.initial' => com.solacesystems.jndi.SolJNDIInitialContextFactory
+         'java.naming.security.principal' => solace-cloud-client@user
+         'java.naming.provider.url' => tcp://address.messaging.solace.cloud:20608
+         'java.naming.security.credentials' => verysecret
+        } <3>
+        require_jars=> ['/usr/share/jms/commons-lang-2.6.jar',
+                        '/usr/share/jms/sol-jms-10.5.0.jar',
+                        '/usr/share/jms/geronimo-jms_1.1_spec-1.1.1.jar',
+                        '/usr/share/jms/commons-lang-2.6.jar]' <4>
+    }
+ }
+--------------------------------------------------
+
+<1> Configuration settings. Note that there is no `broker_url` or `username` or `password` defined here. These are
+    defined in the `jndi_context` hash.
+<2> Username and password for the connection.
+<3> JNDI context settings hash. Contains details of how to connect to JNDI server. See your JMS provider documentation for
+     implementation specific details.
+<4> List of jars required by the JMS provider. Paths should be the fully qualified location of all jar files required
+     by the JMS provider. This list may also include dependent jars as well as specific jars from the JMS provider.
+
+
+===== Sample Configuration for Non-JNDI Connection using Logstash Configuration and Yaml file
+
```

Comment:
I will need to do more testing, but I don't think that stacked headings without content between them works.

Reply:
I can add some explanatory content between the headings to fix that